staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -94,7 +95,28 @@ type Pool struct {
 	// by the controller.
 	Generation int64
 
-	// Device names must be unique inside the pool.
+	// Slices is a list of all ResourceSlices that the driver
+	// wants to publish for this pool. The driver must ensure
+	// that each resulting slice is valid. See the API
+	// definition for details, in particular the limit on
+	// the number of devices.
+	//
+	// If slices are not valid, then the controller will
+	// log errors produced by the apiserver.
+	//
+	// Drivers should publish at least one slice for each
+	// pool that they normally manage, even if that slice
+	// is empty. "Empty pool" is different from "no pool"
+	// because it shows that the driver is up-and-running
+	// and simply doesn't have any devices.
+	Slices []Slice
+}
+
+// +k8s:deepcopy-gen=true
+
+// Slice is turned into one ResourceSlice by the controller.
+type Slice struct {
+	// Devices lists all devices which are part of the slice.
```

Comment:
Do we need to indicate the 128 limit here? Because resource slices do not check the relevant limit now, the upper-level Pool needs to do the relevant verification.

Reply:
It's mentioned above:

         ... The driver must ensure
	// that each resulting slice is valid. See the API
	// definition for details, in particular the limit on
	// the number of devices.
	
I don't want to repeat here what "valid" means because such documentation can easily become stale.

Reply:
I haven't gotten deep enough in the code et to see, but I'm assuming that attempting to write a slice that has more then 128 devices will return an error, correct? I.e we will attempt to write it to the API server, it will fail validation, and then `PublishResources()` will fail, correct?